testutil/daemon/daemon.go

```diff
@@ -621,6 +621,11 @@ func (d *Daemon) Kill() error {
 		return err
 	}
 
+	_, err := d.cmd.Process.Wait()
```

Comment:
Will this affect other tests that don't require it to be gone  (ie should this be optional?)

Reply:
SIGKILL is sent to the daemon process by this method. So I don't see how other tests couldn't require it not to be gone.

Reply:
Yeah, maybe it's fine; the case I had in mind would be a test spinning up a daemon, killing one, and creating a new one (without caring about the previous one - not sure if the wait would cause our CI to take longer)

Reply:
If another daemon is started with the exact same config (eg. same data dirs), there's a short window of time where both daemons might collide. So overall, even if it adds a few ms of delay, I think it's a net gain. I even removed a `time.Sleep` that was presumably added to avoid that sort of bug.